xdsl/__init__.py

```diff
@@ -0,0 +1,12 @@
+from .ir import Operation
+from .builder import Builder
+
+# set up the operation callback for implicit construction
+
+
+def _op_init_callback(op: Operation):
+    if (b := Builder.get_implicit_builder()) is not None:
```

Comment:
This is run on every single operation instantiation, right? And if there is an implicit builder, it will be built there?

Reply:
No, this is run when you import anything from the `xdsl` module, only once per python process starting

Reply:
Yes, and this sets a callback in the operation class. This callback will be run on each instantiation, right?

Reply:
Yes

Reply:
hmm. So it's more the implementation that the interface that you would like to change?

Reply:
Yes. I think the proposed API makes sense for the usecase you seem to have in mind, but I fear that the current implementation might make other parts of the repo depend on this part in a nontrivial way.

Reply:
Yeah it's a good point, and I generally agree. I'll have a go at an implementation with AST transformations.

Reply:
I've removed the callback, but kept most of the rest of the mechanism the same. The onion is intact, and the operation doesn't know about the builder or accommodate it at all.